Modules/xSharePoint/DSCResources/MSFT_xSPOutgoingEmailSettings/MSFT_xSPOutgoingEmailSettings.psm1

```diff
@@ -0,0 +1,94 @@
+function Get-TargetResource
+{
+	[CmdletBinding()]
+	[OutputType([System.Collections.Hashtable])]
+	param
+	(
+		[parameter(Mandatory = $true)] [System.String] $WebAppUrl,
+		[parameter(Mandatory = $true)] [System.String] $SMTPServer,
+		[parameter(Mandatory = $true)] [System.String] $FromAddress,
+		[parameter(Mandatory = $true)] [System.String] $ReplyToAddress,
+		[parameter(Mandatory = $true)] [System.String] $CharacterSet
+		
+	)
+
+	Write-Verbose -Message "Retrieving outgoing email settings configuration "
+
+	$result = Invoke-xSharePointCommand -Arguments $PSBoundParameters -ScriptBlock {
+		$params = $args[0]
+		
+		$webApp = $null
+		if([string]::IsNullOrEmpty($params.WebAppUrl)){
```

Comment:
another interesting question. This is a sort of left over I left from
before I learnt how to run in DSC mode :) Idea was to look for CA when
$webAppUrl = "".

i just removed the unnecessary ifs.

It does mean people need to be aware GLOBAL outgoing email settings is
actually a configuration set to the CA Web App.







On Sun, Nov 15, 2015 at 10:53 PM, Brian Farnhill <notifications@github.com>
wrote:

> In
> Modules/xSharePoint/DSCResources/MSFT_xSPOutgoingEmailSettings/MSFT_xSPOutgoingEmailSettings.psm1
> <https://github.com/PowerShell/xSharePoint/pull/95#discussion_r44870783>:
>
> > +	(
> > +		[parameter(Mandatory = $true)] [System.String] $WebAppUrl,
> > +		[parameter(Mandatory = $true)] [System.String] $SMTPServer,
> > +		[parameter(Mandatory = $true)] [System.String] $FromAddress,
> > +		[parameter(Mandatory = $true)] [System.String] $ReplyToAddress,
> > +		[parameter(Mandatory = $true)] [System.String] $CharacterSet
> > +		
> > +	)
> > +
> > +	Write-Verbose -Message "Retrieving outgoing email settings configuration "
> > +
> > +	$result = Invoke-xSharePointCommand -Arguments $PSBoundParameters -ScriptBlock {
> > +		$params = $args[0]
> > +		
> > +		$webApp = $null
> > +		if([string]::IsNullOrEmpty($params.WebAppUrl)){
>
> If WebAppUrl is the key value for this resource, what's the scenario where
> it can be null or empty? I'll test it in my lab but I'm not sure you'll be
> able to pass an empty WebAppUrl value in a configuration
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/PowerShell/xSharePoint/pull/95/files#r44870783>.
>